app/controllers/v0/sessions_controller.rb

```diff
@@ -97,5 +94,16 @@ def async_create_evss_account(user)
       auth_headers = EVSS::AuthHeaders.new(user).to_h
       EVSS::CreateUserAccountJob.perform_async(auth_headers)
     end
+
+    def location_param
+      # Regex explanation:
+      # - must begin with forward-slash: /
+      # - allowable chars: a-z, A-Z, 0-9, _, -, /
+      %r{\A\/[a-zA-Z0-9\/_-]+\z}.match(params[:location])&.to_s || '/profile'
```

Comment:
This is some minimal input validation on the location param.  I think a more secure (but possibly less user friendly) solution would be to have a map of all possible destinations, ex 
```ruby
{'rx' => '/prescriptions', 'sm' => '/secure_messaging'...}
```
and then whitelist the `?location` query param against this map.